edx_proctoring/api.py

```diff
@@ -1186,6 +1198,19 @@ def _get_practice_exam_view(exam, context, exam_id, user_id, course_id):
         student_view_template = 'proctored_exam/ready_to_start.html'
     elif attempt_status == ProctoredExamStudentAttemptStatus.error:
         student_view_template = 'practice_exam/error.html'
+    elif attempt_status == ProctoredExamStudentAttemptStatus.waiting_for_app_timeout:
+        if attempt['last_poll_timestamp'] is not None and \
+                (datetime.now(pytz.UTC) - attempt['last_poll_timestamp']).total_seconds() > \
+                SOFTWARE_SECURE_CLIENT_TIMEOUT:
+            update_attempt_status(
+                exam_id,
+                user_id,
+                ProctoredExamStudentAttemptStatus.submitted
+            )
+            attempt['status'] = ProctoredExamStudentAttemptStatus.submitted
+            student_view_template = 'practice_exam/submitted.html'
+        else:
+            student_view_template = 'proctored_exam/waiting_for_app_timeout.html'
```

Comment:
This seems like duplicate code compared to what is in _get_proctored_exam_view(). Can we make s shared helper method?